### DISCUSSION TITLE
Why using `import * as React from 'react'` ?

### DISCUSSION THREAD

**Discussion by ImBIOS** [shadcn-ui/ui#1819] [Q&A] [answered]
The `import * as React from 'react'` syntax is more than just another way to import React into your project. This discussion delves into the underlying mechanics, advantages, and trade-offs of using this approach, especially in TypeScript projects. We'll explore how this syntax affects tree-shaking, type inference, and compatibility with various build systems. Whether you're a React novice or a seasoned veteran, join the conversation to weigh in on whether this import style is just a fad or an essential tool for modern React development.

**Comment by Aghaei-Dev** [comment]
i never use that syntax 
i just use this after these updates 

```js
import {useRef,useState} from 'react'
```

**Reply by ImBIOS** [reply]
But this repo and radix-ui are using it, right? That's my context, why is it like that over the destructured one?

**Comment by fahmiidris** [comment, marked as answer]
https://theodorusclarence.com/shorts/react/namespace-vs-named you can read this article
